src/config.rs

```diff
@@ -288,30 +287,24 @@ impl<'a> Cfg<'a> {
         let default_host_triple =
             settings_file.with(|s| Ok(get_default_host_triple(s, process)))?;
         // Environment override
-        let env_override = process
-            .var("RUSTUP_TOOLCHAIN")
-            .ok()
-            .and_then(utils::if_not_empty)
+        let env_override = non_empty_var("RUSTUP_TOOLCHAIN", process)?
             .map(ResolvableLocalToolchainName::try_from)
             .transpose()?
             .map(|t| t.resolve(&default_host_triple))
             .transpose()?;
 
-        let dist_root_server = match process.var("RUSTUP_DIST_SERVER") {
-            Ok(s) if !s.is_empty() => {
+        let dist_root_server = match non_empty_var("RUSTUP_DIST_SERVER", process)? {
+            Some(s) => {
                 trace!("`RUSTUP_DIST_SERVER` has been set to `{s}`");
                 s
             }
-            _ => {
+            None => {
                 // For backward compatibility
-                process
-                    .var("RUSTUP_DIST_ROOT")
-                    .ok()
-                    .and_then(utils::if_not_empty)
+                non_empty_var("RUSTUP_DIST_ROOT", process)?
                     .inspect(|url| trace!("`RUSTUP_DIST_ROOT` has been set to `{url}`"))
-                    .map_or(Cow::Borrowed(dist::DEFAULT_DIST_ROOT), Cow::Owned)
                     .as_ref()
```

Comment:
Nope, we still need it. Otherwise the `map()` takes ownership of the value that was previously contained in the `Option`, but it only yields a reference to that value.